### PR TITLE
MRG, FIX: Datetime call in gdf 2.x age calculation

### PR DESCRIPTION
**Pull request by SveaMeyer13**
#### Reference issue
WIth regards to  #7124 .

#### What does this implement/fix?
While importing a GDF 2.x file with

```python
from mne import io
io.read_raw_gdf(fname, preload=True)
```

we encountered the following error:
 
```python
Extracting EDF parameters from /home/svea/Desktop/mne_examples/sample_data/B0103T.gdf...
GDF file detected
Traceback (most recent call last):
  File "/home/svea/Desktop/mne_examples/online_linear_demo.py", line 17, in <module>
    raw = io.read_raw_gdf(raw_fname, preload=True)
  File "/home/svea/Desktop/mne_examples/venv/lib/python3.7/site-packages/mne/io/edf/edf.py", line 1327, in read_raw_gdf
    verbose=verbose)
  File "<decorator-gen-164>", line 21, in __init__
  File "/home/svea/Desktop/mne_examples/venv/lib/python3.7/site-packages/mne/io/edf/edf.py", line 189, in __init__
    exclude, preload)
  File "/home/svea/Desktop/mne_examples/venv/lib/python3.7/site-packages/mne/io/edf/edf.py", line 377, in _get_info
    edf_info, orig_units = _read_header(fname, exclude)
  File "/home/svea/Desktop/mne_examples/venv/lib/python3.7/site-packages/mne/io/edf/edf.py", line 366, in _read_header
    return _read_gdf_header(fname, exclude), None
  File "/home/svea/Desktop/mne_examples/venv/lib/python3.7/site-packages/mne/io/edf/edf.py", line 865, in _read_gdf_header
    today = datetime.today(tzinfo=timezone.utc)
TypeError: today() takes no keyword arguments
```

After closer investigation, we suspect that it was caused by the changes introduced by #7124 and released in v0.20.0.

As stated in the [Python docs for datetime](https://docs.python.org/3.7/library/datetime.html#datetime.datetime.now),
datetime.now(tz=timezone.utc) should be the syntactically correct functional equivalent.

To then make today and patient['birthday'] comparable, we add tzinfo to birthday (in line 860).

### CODE REVIEW

**Comment by cbrnr** [comment]
#7582 duplicate?

**Comment by larsoner** [comment]
Indeed, @SveaMeyer13 can you see if #7582 also works? I slightly prefer that one since it includes a test

**Comment by larsoner** [comment]
@SveaMeyer13 actually, instead can you update `latest.inc` with an update about the bugfix (you can take the wording from my PR if you want -- plus change to your name and add yourself to `names.inc` if not there already) and we can merge this, then I can rebase my PR to get the test in

**Comment by SveaMeyer13** [comment]
Done!

File: doc/changes/names.inc [hunk 1]
```diff
@@ -298,4 +298,6 @@
 
 .. _Adam Li: http://github.com/adam2392
 
-.. _Ramiro Gatti: https://github.com/ragatti
\ No newline at end of file
+.. _Ramiro Gatti: https://github.com/ragatti
+
+.. _Svea Marie Meyer: https://github.com/SveaMeyer13
```
Comment (larsoner) [inline doc/changes/names.inc:303]: Name here and above need to match, it's what CircleCI complains about (search "warning" in the log and you'll see it)

https://app.circleci.com/pipelines/github/mne-tools/mne-python/897/workflows/179f9067-911b-4472-b7eb-20702ffa91f8/jobs/19252

And the broken link is visible here

https://19252-1301584-gh.circle-artifacts.com/0/dev/whats_new.html

**Comment by larsoner** [comment]
Thanks @SveaMeyer13 !

**Pull request merged**